src/datasources/notebook/QueryEditor.tsx

```diff
@@ -47,7 +47,7 @@ export class QueryEditor extends PureComponent<Props, State> {
       this.setState({ notebooks, loadingNotebooks: false });
 
       if (this.props.query.id) {
-        const notebook = this.getNotebook(this.props.query.id);
```

Comment:
There are other places in this file where we call `this.getNotebook` - is this not an issue in those places as well?

Reply:
In "this.getNotebook" we use the state (this.state.notebooks). In this function we set the actual state (at line 47), and the state is not synchronously updated.